.github/mergify.yml

```
@@ -0,0 +1,43 @@
+pull_request_rules:
+- actions:
+    merge:
+      strict: smart
+      method: merge
```

Prefer squash if that's available?

---

we were using clasic merge as default but i am not against it
i think @alanz (among others?) preferred merge though

---

I would be interested in hearing the reasons. `squash` is what the ghcide repo used and it worked very well there

---

Reason has been to preserve the actual process of development.

But I have come to not have a firm opinion on this issue, and on a big repo like this, with the number of changes coming through, squash might actually be a better option.

---

yeah, you can bisect master in fine grain mode without having to jump to the pr's branches, that might be removed (so you have to restore them first)
but personally i am no doing that clase of git history analysis habitually so 🤷 
```suggestion
      method: squash
```